imap_processing/swe/l2/swe_l2.py

```diff
@@ -251,13 +362,96 @@ def swe_l2(l1b_dataset: xr.Dataset, data_version: str) -> xr.Dataset:
+    # Calculate spin phase using SWE acquisition_time from the
+    # L1B dataset. The L1B dataset stores acquisition_time with
+    # dimensions (epoch, esa_step, spin_sector). Use center time
+    # to calculate spin phase. This center time calculation is
+    # necessary to accurately determine the center angle of the data.
+    #
+    # To determine the center acquisition time, we adjust the
+    # recorded acquisition_time as follows:
+    #   acquisition_time + (acq_duration / 1000000) / 2
+    #
+    # Here, acq_duration is given in microseconds and is stored
+    # in the L1B dataset with dimensions (epoch, cycle). Since acq_duration
+    # remains the same for all quarter cycles within a full sweep,
+    # we use the first acq_duration value for each full sweep to perform
+    # this adjustment.
+
+    acq_duration = l1b_dataset["acq_duration"].data[:, 0] / 2000000
+    data_acq_time = (
+        l1b_dataset["acquisition_time"].data + acq_duration[:, np.newaxis, np.newaxis]
+    )
+
+    # Calculate spin phase
+    inst_spin_phase = get_instrument_spin_phase(
+        query_met_times=data_acq_time.flatten(),
+        instrument=SpiceFrame.IMAP_SWE,
+    )
+
+    # Convert spin phase to spin angle in degrees.
+    inst_spin_angle = get_spin_angle(inst_spin_phase, degrees=True).reshape(-1, 24, 30)
+
+    # Save spin angle in dataset per SWE request.
+    dataset["inst_az_spin_sector"] = xr.DataArray(
+        inst_spin_angle,
+        name="inst_az_spin_sector",
+        dims=["epoch", "energy", "inst_az"],
+        attrs=cdf_attributes.get_variable_attributes("inst_az_spin_sector"),
+    )
+
+    # The spin angle bins are centered at:
+    #   [ 6, 18, 30, 42, 54, 66, 78, 90, 102, 114, 126, 138, 150, 162, 174,
+    #     186, 198, 210, 222, 234, 246, 258, 270, 282, 294, 306, 318, 330,
+    #     342, 354]
+    #
+    # An input angle is assigned to a bin based on the following conditions:
+    #   - phi_begin <= center - 6
+    #   - phi_center = 6
+    #   - phi_end < center + 6
+    #
+    # For example, if the input angle is 8.4, it falls within the bin centered at 6.
+    #
+    # To make binning easier, we define bin edges as:
+    #   [0, 12, 24, 36, 48, 60, 72, 84, 96, 108, 120, 132, 144, 156, 168,
+    #    180, 192, 204, 216, 228, 240, 252, 264, 276, 288, 300, 312, 324,
+    #    336, 348]
+    #
+    # SWE uses the right-side behavior of `np.searchsorted`, where `a[i-1] <= v < a[i]`.
+    #
+    # Example test cases:
+    #   - `np.searchsorted(x, [6], side="right") -> [1]` (Bin center test)
+    #   - `np.searchsorted(x, [8.4], side="right") -> [1]` (Edge case near center)
+    #   - `np.searchsorted(x, [12], side="right") -> [2]` (Bin end test)
+    #   - `np.searchsorted(x, [0], side="right") -> [1]` (Bin start test)
+    #
+    # Using `i-1` ensures that all input angles are assigned to the correct bin of
+    # centered angle bins.
+
+    spin_angle_bins_range = np.arange(0, 360, 12)
```

Comment:
I suggest renaming this to be consistent with your above comment as well as histogram nomenclature.
```suggestion
    spin_angle_bin_edges = np.arange(0, 360, 12)
```